specification.md

```diff
@@ -0,0 +1,89 @@
+# OpenTelemetry/Prometheus Compatibility Specification
+
+Status: [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/document-status.md)
+
+## Abstract
+
+OpenTelemetry is aiming to provide compatibility with
+Prometheus and OpenMetrics. This document explains the
+extent of the support. OpenTelemetry collector and libraries
+will align with the compatibility requirements defined in
+this spec.
+
+## Goals
+
+* OpenTelemetry collector can be used as a drop-in replacement
+  for Prometheus server to scrape and export metrics data.
+* OpenTelemetry collector should export OLTP-compatible Prometheus
+  time series to OpenTelemetry metrics exporters.
+* OpenTelemetry libraries should implement exporters to publish
```

Comment:
Does this include in-process endpoints the enable Kubernetes to use OTel?

Reply:
Is there an example of an in-process endpoint you can share? I'm not very familiar with in-process Prometheus endpoints.

Reply:
@brancz could you give context here? I don't want to say something subtly wrong.

Reply:
Just that the Kubernetes processes (kubelet, apiserver, etc.) must have the `/metrics` endpoint themselves and they must work without external components. Kubernetes has stability guarantees on these endpoints as well as some of the metrics contained in them which is not something that can be broken.

Just a side note though, in the last sig-instrumentation meeting there were various voices that it may not be feasible at all for Kubernetes to move away from prometheus/client_golang as Kubernetes makes a lot of use of multiple registries for multiple /metrics endpoints and uses custom collectors all over the place.

Reply:
So far I believe we've been discussing Prometheus compatibility at the server and data-model level, not in the client API (@reyang take note). OTel SDKs will certainly support a `/metrics` endpoint, and OTel has concepts that enable different registries (separate SDKs). I believe that "custom collector" in OTel client terms translates into "asynchronous instrument" in the OTel model API.

OTel has not stated a compatibility goal to offer drop-in compatible clients for metrics the way it did for OpenTracing in the tracing area; it would be an interesting exercise to try to take the OTel-Go Metrics SDK and implement the Prometheus client API including custom collectors. I think it's possible, but would be a serious effort, and there would still be fine-level distinctions worth pointing out (e.g., we have a difference in how histograms are synchronized presently in these two clients).

Reply:
Yeah, to be honest, for Kubernetes I'm not even entirely sure it's worth the effort to create a compatibility API, if sig-instrumentation deems it worth it to move to otel then putting in the work of migrating is probably better anyways which you'd eventually want even with a go compatible API. If we have the "a prometheus and otel target are indistinguishable to the collector" characteristic, then Kubernetes is covered by otel either way, I think that's all people really care about most. I just wanted to share it.

Reply:
As you mentioned, the current focus and scope is the collector. Let's discuss the requirements from the SDKs more in detail when we start working on the Phase 2.